src/Symfony/Component/DomCrawler/AbstractUriElement.php

```diff
@@ -18,20 +18,28 @@
  */
 abstract class AbstractUriElement
 {
+    /**
+     * @deprecated since Symfony 7.1, use `$domeNode` instead
+     */
     protected \DOMElement $node;
+    protected \DOMElement|\DOM\Element $domNode;
     protected ?string $method;
     protected ?string $currentUri;
 
     /**
-     * @param \DOMElement $node       A \DOMElement instance
+     * @param \DOMElement|\DOM\Element $node       A \DOMElement or a \DOM\Element instance
      * @param string|null $currentUri The URI of the page where the link is embedded (or the base href)
      * @param string|null $method     The method to use for the link (GET by default)
      *
      * @throws \InvalidArgumentException if the node is not a link
      */
-    public function __construct(\DOMElement $node, ?string $currentUri = null, ?string $method = 'GET')
+    public function __construct(mixed $node, ?string $currentUri = null, ?string $method = 'GET')
```

Comment:
Unfortunately, I didn't find another way to do this properly. We could indeed create a new class, like `AbstractDomUriElement` but I feel like it would not be really worth it. Typing `mixed` prevents any BC break, and the type-hint will be restored in Symfony 8.0.

Reply:
Use a union type instead.

Reply:
But this class is not internal. That was what I wanted to do but adding a new type with a union will break BC, isn't it?

Reply:
Widening a parameter type in the constructor is not a BC break (and your change going to `mixed` is actually widening it even more than the union type)